plugin/jsgf.vim

```diff
@@ -10,39 +10,36 @@ let loaded_jsgf_plugin = 1
 function! InitJSGF()
   setlocal suffixesadd+=.js,.vue,.json,.jsx,.ts,.tsx
   setlocal isfname+=@-@
-  " setlocal includeexpr=v:fname.'/index'
   let node_modules = finddir('node_modules', expand('%:p:h') . ';')
   execute 'setlocal path+=' . node_modules
-  " setlocal path+=node_modules
-  " let project_root=findfile('package.json', expand('%:p:h') . ';')
-  " execute 'setlocal path+=' . fnamemodify(project_root, ':p:h') . '/node_modules'
 endfunction
 
 function! FindFileOrDir(filename)
-  let filenames = [
-    \ a:filename,
-    \ a:filename . '.js',
-    \ a:filename . '.vue',
-    \ a:filename . '.json',
-    \ a:filename . '.jsx',
-    \ a:filename . '.ts',
-    \ a:filename . '.tsx'
-  \ ]
-  for fname in filenames
-    if filereadable(fname)
-      return fname
+  if filereadable(a:filename)
+    return a:filename
+  endif
+
+  let suffixes = split(&suffixesadd, ',')
+  let suffixes = filter(suffixes, 'count(suffixes, v:val) == 1')
+
+  for s in suffixes
+    let candidate = a:filename . s
+    if filereadable(candidate)
+      return candidate
     endif
   endfor
   return ''
 endfunction
 
 function! JSGF(filepath, open)
   let filename = a:filepath
+
   if isdirectory(filename)
     let pkg_file = filename . '/package.json'
-
-    if filereadable(pkg_file)
-
+    if (FindFileOrDir(filename . '/index') != '')
```

Comment:
can't change this condition order.

when dir match, but with a module in the dir, this way will be wrong.